internal/events/cmdcall/cmdcall.go

```diff
@@ -0,0 +1,104 @@
+package cmdcall
+
+import (
+	"strings"
+
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/scriptrun"
+	"github.com/ActiveState/cli/internal/subshell"
+	"github.com/ActiveState/cli/pkg/project"
+)
+
+type primeable interface {
+	primer.Outputer
+	primer.Projecter
+	primer.Subsheller
+}
+
+// CmdCall manages dependencies for the handling of events triggered by command
+// calls.
+type CmdCall struct {
+	out       output.Outputer
+	proj      *project.Project
+	subshell  subshell.SubShell
+	cmdList   string
+	p         primeable
+	scriptrun *scriptrun.ScriptRun
+}
+
+// New returns a prepared pointer to an instance of CmdCall.
+func New(p primeable, cmdList string) *CmdCall {
+	return &CmdCall{
+		out:       p.Output(),
+		proj:      p.Project(),
+		subshell:  p.Subshell(),
+		cmdList:   cmdList,
+		p:         p,
+		scriptrun: scriptrun.New(p.Output(), p.Subshell(), p.Project()),
+	}
+}
+
+// Run executes the event handling logic by running any relevant scripts.
+func (cc *CmdCall) Run(eventType project.EventType) error {
```

Comment:
Please include the event name in your localized errors here. Localized error should ideally be actionable. The current errors are very vague and won't tell the user much.